src/lib/pipeline/ReorderBuffer.cc

```diff
@@ -81,6 +81,20 @@ unsigned int ReorderBuffer::commit(uint64_t maxCommitSize) {
   unsigned int n;
   for (n = 0; n < maxCommits; n++) {
     auto& uop = buffer_[0];
+    if (uop->getInstructionAddress() == last_inst_addr) {
+      inst_repeat_counter++;
+    } else {
+      inst_repeat_counter = 0;
+    }
+    if (inst_repeat_counter > 10000000) {
```

Comment:
Definitely worth a brief discussion. Perhaps worth a config option that defaults to a large value (10 or 100 million cycles) that we don't explicitly include in the provided configurations. Just in case someone wants to try out some crazy memory backends with huge latencies. If necessary, I can always experiment with the longest no. cycles an instruction in a bottlenecked CPU (slow memory, small frontend) stays at the head of the ROB, though I'd expect this is for a few thousand cycles and nowhere near 10 million.